[flutter_tools] - Add `queries` section to Android manifest file

bleroux opened the pull request:
## Description

This PR adds a new section to the Android manifest file.
This section is required for https://github.com/flutter/engine/pull/44579 (because it uses `PackageManager.queryIntentActivities` API ), see:

1. https://developer.android.com/training/package-visibility?hl=en
2. https://developer.android.com/reference/android/content/Intent#ACTION_PROCESS_TEXT
3. https://developer.android.com/reference/android/content/pm/PackageManager?hl=en#queryIntentActivities(android.content.Intent,%20android.content.pm.PackageManager.ResolveInfoFlags)

## Related Issue

Android manifest update for https://github.com/flutter/flutter/issues/139361.

## Tests

This PR updates the integration tests and examples Android manifest files, this will help catch error or warning related to this change.

bleroux committed:
[tools] - Add `queries` section to Android manifest file

bleroux commented:
This is WIP to get feedbacks:
- does such template change require a test?
- does this change should be applied to existing Android manifest files in the repo (for instance, `/Users/bruno/Nevercode/flutter/dev/integration_tests/channels/android/app/src/main/AndroidManifest.xml`)?
- if it makes sense, is there a way to get this section being automatically to an existing Flutter project?

camsim99 commented:
> This is WIP to get feedbacks:
> 
> 1. does such template change require a test?
> 2. does this change should be applied to existing Android manifest files in the repo (for instance, `/Users/bruno/Nevercode/flutter/dev/integration_tests/channels/android/app/src/main/AndroidManifest.xml`)?
> 3. if it makes sense, is there a way to get this change being automatically applied to an existing Flutter project?

1. I don't believe so! But:
2. It would be nice I think the add the new section to integration tests that we run in CI at least. @reidbaker do you have a strong opinion on this though?
3. I'm not sure that there would be besides asking folks to re-create (via `flutter create`) their project to get the effects of the updated template. But, hoping Reid may have some further insight here, as well.

reidbaker commented:
> > This is WIP to get feedbacks:
> > 
> > 1. does such template change require a test?
> > 2. does this change should be applied to existing Android manifest files in the repo (for instance, `/Users/bruno/Nevercode/flutter/dev/integration_tests/channels/android/app/src/main/AndroidManifest.xml`)?
> > 3. if it makes sense, is there a way to get this change being automatically applied to an existing Flutter project?
> 
> 1. I don't believe so! But:
> 2. It would be nice I think the add the new section to integration tests that we run in CI at least. @reidbaker do you have a strong opinion on this though?
> 3. I'm not sure that there would be besides asking folks to re-create (via `flutter create`) their project to get the effects of the updated template. But, hoping Reid may have some further insight here, as well.

Yes I would like for this pr to modify at least one example app to have this permission and ensure it builds correctly and does not add any new android/lint warnings. Also if code is added later that would break we at least have a test to work off of. 

The engine pr appears to have landed and seems to have a safe default. You could write a migration if you/we thought every app needed to update but I think this is more specialized. I would encourage you to add some documentation for developers that want the same functionality but already have an existing app so they know what parts they need to adopt or modify to get the new engine code working.

bleroux committed:
Update integration tests and examples manifest

bleroux committed:
Add documentation about the <queries> element

bleroux commented:
@camsim99 @reidbaker Thanks for the feedbacks.

I updated this PR:
- added the `<queries>` element to integration tests and examples manifest files.
- updated the documentation for `DefaultProcessTextService` to mention that the `<queries>` element is required to get a non-empty result.
